src/Sylius/Bundle/CoreBundle/Applicator/CatalogPromotionApplicator.php

```diff
@@ -13,19 +13,30 @@
 
 namespace Sylius\Bundle\CoreBundle\Applicator;
 
+use Sylius\Component\Core\Model\CatalogPromotionInterface;
 use Sylius\Component\Core\Model\ChannelPricingInterface;
 use Sylius\Component\Core\Model\ProductVariantInterface;
 
 final class CatalogPromotionApplicator implements CatalogPromotionApplicatorInterface
 {
-    public function applyPercentageDiscount(ProductVariantInterface $variant, float $discount): void
-    {
+    public function applyCatalogPromotion(
+        ProductVariantInterface $variant,
+        CatalogPromotionInterface $catalogPromotion
+    ): void {
         /** @var ChannelPricingInterface $channelPricing */
         foreach ($variant->getChannelPricings() as $channelPricing) {
             if ($channelPricing->getOriginalPrice() === null) {
                 $channelPricing->setOriginalPrice($channelPricing->getPrice());
             }
-            $channelPricing->setPrice((int) ($channelPricing->getPrice() - ($channelPricing->getPrice() * $discount)));
+
+            $channelPricing->setPrice((int) ($channelPricing->getPrice() - ($channelPricing->getPrice() * 0.5)));
```

Comment:
0.5 🤔 just call me when you want to drink 🍺 
Jokes aside - why 0.5? Is it hardcoded at the moment?

Reply:
It was hardcoded even before and it will be hardcoded until we provide some basic actions for catalog promotions (which is already in progress https://github.com/Sylius/Sylius/pull/13029) 🖖

Reply:
actions has been merged :)

Reply:
But not their application (yet) xD https://github.com/Sylius/Sylius/pull/13036 (but it will probably be merged before and then I change it 🖖)